pkg/backup/item_backupper.go

```diff
@@ -579,8 +578,8 @@ func (ib *itemBackupper) takePVSnapshot(obj runtime.Unstructured, log logrus.Fie
 	return kubeerrs.NewAggregate(errs)
 }
 
-func (ib *itemBackupper) checkResourcePolicies(obj runtime.Unstructured, groupResource *schema.GroupResource, actionName string) (*resourcepolicies.Action, error) {
-	if ib.backupRequest.ResPolicies != nil && groupResource.String() == "persistentvolumeclaims" && actionName == "velero.io/csi-pvc-backupper" {
+func (ib *itemBackupper) matchResourcePolicies(obj runtime.Unstructured, groupResource schema.GroupResource, backupItemActionName string) (*resourcepolicies.Action, error) {
```

Comment:
Trivial, but
I don't quite understand why this func is needed.  The struct `Polices` should be extensible so it can match all kinds of resources.
And if the `match` function returns `skip` action you should check the name in the for-loop of triggering BIA rather than this func.

such that his func is unnecessary?

Reply:
our `Policies` struct has the function [MatchResourcePolicies](https://github.com/vmware-tanzu/velero/blob/c93454019024ac97dec92bb58a7c9d185295cf3a/internal/resourcepolicies/resource_policies.go#L103) that could match all kinds of resources. but currently, we only have `PV` and `Pod Volume` to handle. If we have other resources that want to match, we could extend the `MatchResourcePolicies` function.

Here why we need matchResourcePolicies b/c we need to use kubeClient to find PVC-related PV. and then call the MatchResourcePolicies function

Another place is [matchResourcePolicies](https://github.com/vmware-tanzu/velero/blob/c93454019024ac97dec92bb58a7c9d185295cf3a/pkg/backup/item_backupper.go#L581), also it needs to find PVC-related PV, but it used `pvClient corev1client.PersistentVolumesGetter` kinds of client to find PV.
and then call the MatchResourcePolicies function

So it's not easy to uniform interface further. and this's why we have two matchResourcePolicies function in itemBackupper and backupper before we call  MatchResourcePolicies function

Reply:
> Here why we need matchResourcePolicies b/c we need to use kubeClient to find PVC-related PV. and then call the MatchResourcePolicies function

Isn't this a common logic that whenever it needs to filter a PVC it needs to check the PV also?

Reply:
yes, for handling PVC we need to check related PV **through k8s client,** but we have two different clients(kbclient, client-go) currently,  so we could not use one unified function to get matched action for PVC.

So here used add another logic that for PVC we first get related PV, then through PV resource get matched action.